### PR TITLE
Use shared repositories instead of API calls / Add policy api endpoints / Add Kafka Consumers/Producers

### PR DESCRIPTION
**Pull request by jjbuchan**
# What

* monitorId uses UUID instead of String
* Removed resource/monitor api clients
* Add kafka consumer for `PolicyMonitorUpdateEvent`s
* Add kafka producer for policy monitor updates and tenant metadata changes
* Renamed `policyScope` to `scope` - having the class as `PolicyScope` was the initial intended change and this rename helps make things more readable.
* Adds a couple new API endpoints

# How

By moving entities/repositories to salus-model it makes it easy.

# Why

Less hard dependencies between services.

`PolicyMonitorUpdateEvent` will be received here without the tenant field populated.  This will then look up the relevant tenants for the polciy and then send out new events for each tenant to be consumed by MonitorMgmt.

### CODE REVIEW

File: src/main/resources/application-dev.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ salus:
 spring:
   jpa:
     hibernate:
-      ddl-auto: update
+      ddl-auto: create
```
Comment (itzg) [inline src/main/resources/application-dev.yml]: Just curious why this change

Comment (jjbuchan) [reply src/main/resources/application-dev.yml]: This comment made me realize I had some committed changes that weren't pushed up.

Two new commits now that reverts this and includes a few other changes.  I'll update the PR description.

I think I temporarily switched to `create` to see if it'd help that index warning (before you discovered the problem).

File: src/main/java/com/rackspace/salus/policy/manage/services/PolicyManagement.java [hunk 2]
```diff
@@ -199,8 +218,36 @@ private boolean exists(MonitorPolicyCreate policy) {
...
+        .forEach(policyEventProducer::sendPolicyEvent);
+  }
+
+  public void handlePolicyMonitorUpdate(UUID monitorId) {
+    Optional<MonitorPolicy> policy = monitorPolicyRepository.findByMonitorId(monitorId);
+    if (policy.isEmpty()) {
+      log.warn("Ignoring policy monitor update for monitor={}, monitor not used in policy", monitorId);
+      return;
+    }
+    log.info("Sending policy monitor update events for {}", policy.get());
```
Comment (itzg) [inline src/main/java/com/rackspace/salus/policy/manage/services/PolicyManagement.java]: Perhaps debug log level

Comment (jjbuchan) [reply src/main/java/com/rackspace/salus/policy/manage/services/PolicyManagement.java]: I did info here since it will only log once per update, not once per tenant.
